### PR TITLE
WIP: Add support for reading token from SSM

### PR DESCRIPTION
**Pull request by paulolai**
Modify the Lambda function to read the buildkite agent token from SSM, also clean up some vendor packages and switch to the AWS provided lambda libraries.

This is still a WIP - 
1. this has two main definitions
2. I haven't tested that the local version still works.
3. The makefile needs some cleaning up

### CODE REVIEW

**Comment by lox** [comment]
Thanks very much for the PR! 🙌🏻

Perhaps let's get the vendors updated in one PR and then we can look at the SSM changes in isolation?

File: README.md [hunk 1]
```diff
@@ -44,7 +44,7 @@ It's entrypoint is `handler.handle`, it requires a `python2.7` environment and r
  - BUILDKITE_QUIET
  - BUILDKITE_CLOUDWATCH_DIMENSIONS
 
-Take a look at https://github.com/buildkite/elastic-ci-stack-for-aws/blob/v3.3.0/templates/metrics.yml for examples of usage.
```
Comment (uri-canva) [inline README.md]: This change seems unintentional, a merge error perhaps?

File: agent-metrics.tf [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+  schedule_expression = "rate(1 minute)"
+}
+
+resource "aws_lambda_permission" "allow_cloudwatch_to_invoke_lambda" {
+  statement_id = "AllowExecutionFromCloudWatch"
+  action = "lambda:InvokeFunction"
+  function_name = "${aws_lambda_function.buildkite-metrics-function.function_name}"
+  principal = "events.amazonaws.com"
+  source_arn = "${aws_cloudwatch_event_rule.every_minute.arn}"
+}
```
Comment (uri-canva) [inline agent-metrics.tf]: Unrelated.

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+	terraform destroy
+	rm -f init.done deploy.done monitor.zip monitor
+
+init.done:
+	terraform init
+	touch $@
+
+deploy.done: init.done agent-metrics.tf monitor.zip
+	terraform apply
+	touch $@
```
Comment (uri-canva) [inline Makefile]: Unrelated?

File: backend/cloudwatch.go [hunk 4]
```diff
@@ -8,7 +8,7 @@ import (
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/aws/aws-sdk-go/service/cloudwatch"
-	"github.com/buildkite/buildkite-metrics/collector"
+	"github.com/buildkite/buildkite-agent-metrics/collector"
```
Comment (uri-canva) [inline backend/cloudwatch.go]: I'd pull out all the renames of `github.com/buildkite/buildkite-metrics` to `github.com/buildkite/buildkite-agent-metrics` into a separate PR.

File: README.md (hunk 1 above)
Comment (paulolai) [reply README.md]: Is a merge error, will need to fix that up!

File: backend/cloudwatch.go (hunk 4 above)
Comment (paulolai) [reply backend/cloudwatch.go]: I've created https://github.com/buildkite/buildkite-agent-metrics/pull/56

Comment (paulolai) [reply backend/cloudwatch.go]: https://github.com/buildkite/buildkite-agent-metrics/pull/56

File: Makefile (hunk 3 above)
Comment (paulolai) [reply Makefile]: The idea was to add a make file and the terraform file to make it easy to build and deploy the lambda, it may not match up with the rest of the project though.

**Comment by paulolai** [comment]
> Thanks very much for the PR! 🙌🏻
> 
> Perhaps let's get the vendors updated in one PR and then we can look at the SSM changes in isolation?

Welcome.

@lox I'm breaking it down, so far I have the following to do some cleanup and to replace eawsy/aws-lambda-go with aws-lambda-go Please take a look at these PRs:

https://github.com/buildkite/buildkite-agent-metrics/pull/56
https://github.com/buildkite/buildkite-agent-metrics/pull/57
https://github.com/buildkite/buildkite-agent-metrics/pull/58

**Comment by paulolai** [comment]
This is replaced by https://github.com/buildkite/buildkite-agent-metrics/pull/59
